### PR TITLE
Feature/osm zoom to marker

### PR DESCRIPTION
**Pull request by NiclasNorin**

### CODE REVIEW

File: source/js/openstreetmap/helpers/osmHelpers.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        setTimeout(function () {
+            marker.openPopup();
+        }, 300);
+    }
+}
+
+export function createMarkerElementPairs(map, markers, container) {
+    if (!map || !markers || !container) return;
+    const sidebar = container.querySelector('.c-openstreetmap__sidebar');
+    const sidebarItems = sidebar.querySelectorAll('[js-map-lng][js-map-lat]');
```
Comment (thorbrink) [inline source/js/openstreetmap/helpers/osmHelpers.js:14]: sidebar is possibly null.

Comment (thorbrink) [inline source/js/openstreetmap/helpers/osmHelpers.js:14]: sidebar is possibly null.
```suggestion
    const sidebarItems = sidebar?.querySelectorAll('[js-map-lng][js-map-lat]') ?? [];
```

File: source/js/openstreetmap/showPost.js [hunk 2]
```diff
@@ -30,51 +46,63 @@ class ShowPost {
         });
     }
 
+    handleParams() {
```
Comment (thorbrink) [inline source/js/openstreetmap/showPost.js:49]: Could we give this function a less generic name, possibly extract it into smaller functions?

File: source/js/openstreetmap/zoomMarkerClick.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { zoomToMarker } from './helpers/osmHelpers';
+
+class ZoomMarkerClick {
```
Comment (thorbrink) [inline source/js/openstreetmap/zoomMarkerClick.js:3]: Nice! Easy to read this entire class, as well as the size of it :)

File: source/js/openstreetmap/zoomMarkerScroll.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        this.container = container;
+
+        this.init();
+    }
+
+    init() {
+        if (this.scrollToMarkerItems.length <= 0) return;
+        
+        this.setListeners();
+        // console.log(this.scrollToMarkerItems);
```
Comment (thorbrink) [inline source/js/openstreetmap/zoomMarkerScroll.ts:22]:
```suggestion
```

File: source/js/openstreetmap/zoomMarkerScroll.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                this.handleScroll();
+                currentScroll = scrollTop;
+            }
+        });
+    }
+
+
+    handleScroll() {
+        // console.log('scroll');
+    }
```
Comment (thorbrink) [inline source/js/openstreetmap/zoomMarkerScroll.ts:39]: This does nothing.

**Review by thorbrink — CHANGES_REQUESTED** [review]

File: source/js/openstreetmap/zoomMarkerScroll.ts (hunk 4 above)
Comment (NiclasNorin) [reply source/js/openstreetmap/zoomMarkerScroll.ts:22]: This class doesnt get initiated at the moment. Within short we will add scroll functionality to the map, this class is set up as a test to handle this

**Review by thorbrink — APPROVED** [review]
🥇

**Pull request merged**